flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/hooks/TestMasterHook.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.hooks;
+
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+import org.apache.flink.runtime.checkpoint.CheckpointCoordinatorMasterHooksTest;
+import org.apache.flink.runtime.checkpoint.MasterTriggerRestoreHook;
+
+import javax.annotation.Nullable;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * A {@link MasterTriggerRestoreHook} implementation for testing.
+ */
+public class TestMasterHook implements MasterTriggerRestoreHook<String> {
+
+	private static final String DEFAULT_STATE = "default";
+
+	private final String id;
+
+	private int restoreCount = 0;
+
+	private boolean failOnRestore = false;
+
+	public TestMasterHook(String id) {
+		this.id = id;
```

Comment:
I'd prefer a `checkNotNull` here. Because `id` is only used in a getter, it might be that the `NPE` is thrown too late otherwise.